homeassistant/components/transmission.py

```diff
@@ -0,0 +1,193 @@
+        self.data = None
+        self.torrents = None
+        self.available = True
+        self._api = api
+        self.completed_torrents = []
+        self.started_torrents = []
+        self.hass = hass
+
+    def update(self):
+        """Get the latest data from Transmission instance."""
+        from transmissionrpc.error import TransmissionError
+
+        try:
+            self.data = self._api.session_stats()
+            self.torrents = self._api.get_torrents()
+
+            self.check_completed_torrent()
+            self.check_started_torrent()
+
+            _LOGGER.debug("Torrent Data updated")
+            self.available = True
+        except TransmissionError:
+            self.available = False
+            _LOGGER.error("Unable to connect to Transmission client")
+
+    def init_torrent_list(self):
+        """Initialize torrent lists."""
+        self.torrents = self._api.get_torrents()
+        self.completed_torrents = [
+            x.name for x in self.torrents if x.status == "seeding"]
+        self.started_torrents = [
+            x.name for x in self.torrents if x.status == "downloading"]
+
+    def check_completed_torrent(self):
+        """Get completed torrent functionality."""
+        actual_torrents = self.torrents
+        actual_completed_torrents = [
+            var.name for var in actual_torrents if var.status == "seeding"]
+
+        tmp_completed_torrents = list(
+            set(actual_completed_torrents).difference(
+                self.completed_torrents))
+
+        for var in tmp_completed_torrents:
+            self.hass.bus.fire(
+                'transmission_downloaded_torrent', {
+                    'name': var})
+
+        self.completed_torrents = actual_completed_torrents
+
+    def check_started_torrent(self):
+        """Get started torrent functionality."""
+        actual_torrents = self.torrents
+        actual_started_torrents = [
+            var.name for var
+            in actual_torrents if var.status == "downloading"]
+
+        tmp_started_torrents = list(
+            set(actual_started_torrents).difference(
+                self.started_torrents))
+
+        for var in tmp_started_torrents:
+            self.hass.bus.fire(
+                'transmission_started_torrent', {
+                    'name': var})
+        self.started_torrents = actual_started_torrents
+
+    def get_started_torrent_count(self):
+        """Get the number of started torrents."""
+        return len(self.started_torrents)
+
+    def get_completed_torrent_count(self):
+        """Get the number of completed torrents."""
+        return len(self.completed_torrents)
+
+    def set_alt_speed_enabled(self, is_enabled):
+        """Set the alternative speed flag."""
+        self._api.set_session(alt_speed_enabled=is_enabled)
+
+    def get_alt_speed_enabled(self):
+        """Get the alternative speed flag."""
+        return self.get_session().alt_speed_enabled
```

Comment:
We shouldn't have two different interfaces that updates state. Update state in `update` and save that in a cache, ie instance attributes. Then get the state from the cache in the entitites. Then we can remove the throttles.